src/Ansatz.jl

```diff
@@ -305,14 +308,18 @@ function truncate!(::NonCanonical, tn::AbstractAnsatz, bond; threshold, maxdim,
     end
 
     slice!(tn, virtualind, extent)
+    sliced_bond = tensors(tn; bond)
+
+    renormalize && replace!(tn, sliced_bond => sliced_bond ./ norm(tn))
```

Comment:
add comment acknowledging that `normalize!` would be better but can give problems on copied TNs